core/templates/domain/topic/editable-topic-backend-api.service.ts

```diff
@@ -16,180 +16,179 @@
-    };
 
-    var _fetchStories = function(
-        topicId, successCallback, errorCallback) {
-      var storiesDataUrl = UrlInterpolationService.interpolateUrl(
-        TOPIC_EDITOR_STORY_URL_TEMPLATE, {
-          topic_id: topicId
+    this.http.get(topicDataUrl).toPromise().then((response: any) => {
+      if (successCallback) {
+        // The response is passed as a dict with 2 fields and not as 2
+        // parameters, because the successCallback is called as the resolve
+        // callback function in $q in fetchTopic(), and according to its
+        // documentation (https://docs.angularjs.org/api/ng/service/$q),
+        // resolve or reject can have only a single parameter.
+
+        successCallback({
+          topicDict: cloneDeep(response.topic_dict),
+          groupedSkillSummaries: cloneDeep(
+            response.grouped_skill_summary_dicts),
+          skillIdToDescriptionDict: cloneDeep(
+            response.skill_id_to_description_dict),
+          skillIdToRubricsDict: cloneDeep(response.skill_id_to_rubrics_dict)
         });
+      }
+    }, (errorResponse) => {
+      if (errorCallback) {
+        errorCallback(errorResponse);
+      }
+    });
+  }
 
-      $http.get(storiesDataUrl).then(function(response) {
-        var canonicalStorySummaries = angular.copy(
-          response.data.canonical_story_summary_dicts);
-        if (successCallback) {
-          successCallback(canonicalStorySummaries);
-        }
-      }, function(errorResponse) {
-        if (errorCallback) {
-          errorCallback(errorResponse.data);
-        }
+  private _fetchStories(
+      topicId: string, successCallback: any, errorCallback: any): any {
+    var storiesDataUrl = this.urlInterpolation.interpolateUrl(
+      TopicDomainConstants.TOPIC_EDITOR_STORY_URL_TEMPLATE, {
+        topic_id: topicId
       });
-    };
 
-    var _fetchSubtopicPage = function(
-        topicId, subtopicId, successCallback, errorCallback) {
-      var subtopicPageDataUrl = UrlInterpolationService.interpolateUrl(
-        SUBTOPIC_PAGE_EDITOR_DATA_URL_TEMPLATE, {
-          topic_id: topicId,
-          subtopic_id: subtopicId.toString()
-        });
+    this.http.get(storiesDataUrl).toPromise().then((response: any) => {
+      let canonicalStorySummaries = cloneDeep(
+        response.canonical_story_summary_dicts);
+      if (successCallback) {
+        successCallback(canonicalStorySummaries);
+      }
+    }, (errorResponse) => {
+      if (errorCallback) {
+        errorCallback(errorResponse.body);
+      }
+    });
+  }
 
-      $http.get(subtopicPageDataUrl).then(function(response) {
-        var topic = angular.copy(response.data.subtopic_page);
-        if (successCallback) {
-          successCallback(topic);
-        }
-      }, function(errorResponse) {
-        if (errorCallback) {
-          errorCallback(errorResponse.data);
-        }
+  private _fetchSubtopicPage(
+      topicId: string, subtopicId: number, successCallback: any,
```

Comment:
Avoid using the "any" type. Can you please take a look at other spec files to see what type can be used here?